### ISSUE TITLE
Some aria-labelledby attributes have a name instead of a reference to an id

### ISSUE THREAD

**Issue by jobara** [accessibility-in-action/platform#2168]
### Prerequisites

- [X] I've [searched](https://github.com/accessibility-exchange/platform/issues) for existing issues

### Describe the bug

The aria-labelledby attribute should point to an id of an element that will serve as a label. However, in some cases the aria-labelledby is supplied the label value instead.

### To Reproduce

1. Sign in as an org admin
2. Go to an engagement and view the manage participants page
3. Using the web inspector or an accessibility audit tool look at the `aria-labelledby` for the tab navigation elements
4. Notice that it doesn't point at the ID of another element but rather supplies the label itself.

### Expected behavior

The element should be appropriately labelled. Either the aria-labelledby should point at a suitable id, or it should be switched to an aria-label.

### Screenshots

<img width="1226" alt="Screenshot showing the navigation tabs have an invalid aria-labelledby" src="https://github.com/accessibility-exchange/platform/assets/556024/b4e45dd3-0cc0-4c99-8fcd-d35f70679ae5">
